### ISSUE TITLE
AbstractInterval.dimension(d) inconsistent behaviour

### ISSUE THREAD

**Issue by Squareys** [imglib/imglib2#100]
Hello everybody!

I just stumbled over `https://github.com/imglib/imglib2/blob/master/src/main/java/net/imglib2/AbstractInterval.java#L243` throwing an `ArrayIndexOutOfBoundsException`, because I passed a `d >= n`.

In my oppinion, `.dimension()` should return `1`, just as `AbstractImg` does: https://github.com/imglib/imglib2/blob/master/src/main/java/net/imglib2/img/AbstractImg.java#L115

To make behaviour of `IterableInterval` consistent for this use case, I suggest changing the one or the other and *documenting* the behaviour in the `Dimensions` interface.

If you agree, I could do this change myself.

Greetings, 
Squareys

**Comment by tpietzsch** [comment]
@Squareys I agree that it should be consistent. I would prefer to change  ```AbstractImg``` to also throw the exception. What do others think?

**Comment by axtimwalde** [comment]
Agree.

**Comment by dietzc** [comment]
yep. @Squareys can you file a PR?

**Comment by axtimwalde** [comment]
Has @Squareys filed a PR?
